### PR TITLE
fix C4244 warning

### PR DESCRIPTION
**Pull request by Mee-gu**
fix C4244 warning

### CODE REVIEW

File: cocos/2d/CCLabel.cpp [hunk 1]
```diff
@@ -265,7 +266,7 @@ Label* Label::createWithTTF(const std::string& text, const std::string& fontFile
     return nullptr;
 }
 
-Label* Label::createWithTTF(const TTFConfig& ttfConfig, const std::string& text, TextHAlignment hAlignment /* = TextHAlignment::CENTER */, int maxLineWidth /* = 0 */)
+Label* Label::createWithTTF(const TTFConfig& ttfConfig, const std::string& text, TextHAlignment hAlignment /* = TextHAlignment::CENTER */, float maxLineWidth /* = 0 */)
```
Comment (minggo) [inline cocos/2d/CCLabel.cpp:269]: Is it reasonable to use float for label width?

File: cocos/editor-support/cocostudio/CCActionObject.cpp [hunk 2]
```diff
@@ -192,7 +192,7 @@ bool ActionObject::valueToBool(const std::string& value)
 }
 float ActionObject::valueToFloat(const std::string& value)
 {
-    return utils::atof(value.c_str());
+    return (float)utils::atof(value.c_str());
```
Comment (minggo) [inline cocos/editor-support/cocostudio/CCActionObject.cpp:195]: Why need to convert `atof` to float? Isn't `atof` return float?

File: cocos/editor-support/cocostudio/CCArmatureAnimation.cpp [hunk 3]
```diff
@@ -232,7 +232,7 @@ void ArmatureAnimation::play(const std::string& animationName, int durationTo,
         if(movementBoneData && movementBoneData->frameList.size() > 0)
         {
             _tweenList.push_back(tween);
-            movementBoneData->duration = _movementData->duration;
+            movementBoneData->duration = (float)_movementData->duration;
```
Comment (minggo) [inline cocos/editor-support/cocostudio/CCArmatureAnimation.cpp:235]: Why duration have different types?

File: extensions/assets-manager/AssetsManagerEx.cpp [hunk 4]
```diff
@@ -77,7 +77,7 @@ AssetsManagerEx::AssetsManagerEx(const std::string& manifestUrl, const std::stri
                                          int64_t totalBytesReceived,
                                          int64_t totalBytesExpected)
     {
-        this->onProgress(totalBytesExpected, totalBytesReceived, task.requestURL, task.identifier);
+        this->onProgress((double)totalBytesExpected, (double)totalBytesReceived, task.requestURL, task.identifier);
```
Comment (minggo) [inline extensions/assets-manager/AssetsManagerEx.cpp:80]: I think it is more reasonable to modify `onProgress`. Why `onProgress` need to receive double parameters?

File: cocos/editor-support/cocostudio/CCActionObject.cpp (hunk 2 above)
Comment (Mee-gu) [reply cocos/editor-support/cocostudio/CCActionObject.cpp:195]: the `atof` convert string to double.

File: extensions/assets-manager/AssetsManagerEx.cpp (hunk 4 above)
Comment (Mee-gu) [reply extensions/assets-manager/AssetsManagerEx.cpp:80]: I'm afraid that the download size is too big to store in a float.

**Comment by Mee-gu** [comment]
will split into separate PRs

**Pull request closed without merging**